### PR TITLE
add search by ORCIDID method

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made? 🤔

Fixes sul-dlss/happy-heron#3166

1. Add a (temporary implementation) of the search by ORCIDID method that just iterates over all results until it finds what we need and then return it.
2. Update README to indicate how to create new cassettes

## How was this change tested? 🤨

New tests

### CODE REVIEW

File: lib/mais_orcid_client.rb [hunk 1]
```diff
@@ -57,19 +57,51 @@ def fetch_orcid_users(limit: nil, page_size: nil)
...
+  # Fetch a user details,  including scope and token, given either a SUNetID or ORCIDID
+  # @param [string] sunetid of user to fetch
+  # @param [orcid] orcidid of user to fetch (ignored if sunetid is also provided)
+  # @return [<OrcidUser>, nil] orcid user or nil if not found
+  def fetch_orcid_user(sunetid: nil, orcidid: nil)
+    raise "must provide either a sunetid or orcidid" unless sunetid || orcidid
+
+    sunetid ? fetch_by_sunetid(sunetid) : fetch_by_orcidid(orcidid)
+  end
+
```
Comment (peetucket) [inline lib/mais_orcid_client.rb:69]: this existing method is extended to support fetching by sunetid or orcidid

the method signature remains identical for all consumers who search by sunetid, so changes needed for them

File: lib/mais_orcid_client.rb [hunk 2]
```diff
@@ -57,19 +57,51 @@ def fetch_orcid_users(limit: nil, page_size: nil)
...
+        end
+      end
+      next_page = response.dig(:links, :next)
+      return nil if last_page?(response[:links])
+    end
+  end
 
-  def first_page(page_size)
+  def first_page(page_size = nil)
     path = "/users?scope=ANY"
```
Comment (peetucket) [inline lib/mais_orcid_client.rb:105]: make this param optional, and just default to nil, which is supported by the method anyway (it just uses the API default)

File: README.md [hunk 3]
```diff
@@ -58,3 +58,20 @@ end
 After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
 
 To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, update the version number in `version.rb`, and then run `bundle exec rake release`, which will create a git tag for the version, push git commits and the created tag, and push the `.gem` file to [rubygems.org](https://rubygems.org).
+
+## VCR Cassettes
+
+VCR gem is used to record the results of the API calls for the tests.  If you need to
```
Comment (justinlittman) [inline README.md:64]: VCR allows you to obscure automatically through configuration. (See sul_pub for example.)

**Review by justinlittman — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (peetucket) [reply README.md:64]: Yeah, I know - some of the auto obfuscation requires you to specify the string to be obscured, which in sul-pub comes from Settings (which we don't have in the gem).  But I'll check if there is a more clever way to do this, like regex.

**Comment by peetucket** [comment]
Added sanitization where I could, and updated README.  Couldn't figure out how to sanitize multiple mathces of the same regex in a single cassette though (which is needed for user access tokens)

**Review by justinlittman — APPROVED** [review]

**Pull request merged**